nx/source/services/audout.c

```diff
@@ -259,3 +293,103 @@ Result audoutGetAudioOutVolume(float *volume) {
 
     return serviceDispatchOut(&g_audoutIAudioOut, 13, *volume);
 }
+
+Result audoutaRequestSuspendOld(u64 pid, u64 delay, Handle* handle_out) {
+    if (hosversionAtLeast(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    const struct {
+        u64 pid;
+        u64 timespan;
+    } in = { pid, delay };
+
+    return serviceDispatchInOut(&g_audoutaSrv, 0, in, *handle_out);
+}
+
+Result audoutaRequestResumeOld(u64 pid, u64 delay, Handle* handle_out) {
+    if (hosversionAtLeast(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    const struct {
+        u64 pid;
+        u64 timespan;
+    } in = { pid, delay };
+
+    return serviceDispatchInOut(&g_audoutaSrv, 1, in, *handle_out);
+}
+
+Result audoutaRequestSuspend(u64 pid, u64 delay) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    const struct {
+        u64 pid;
+        u64 timespan;
+    } in = { pid, delay };
+
+    return serviceDispatchIn(&g_audoutaSrv, 0, in);
+}
+
+Result audoutaRequestResume(u64 pid, u64 delay) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    const struct {
+        u64 pid;
+        u64 timespan;
+    } in = { pid, delay };
+
+    return serviceDispatchIn(&g_audoutaSrv, 1, in);
+}
+
+Result audoutaGetProcessMasterVolume(u64 pid, float* volume_out) {
+    return serviceDispatchInOut(&g_audoutaSrv, 2, pid, *volume_out);
+}
+
+Result audoutaSetProcessMasterVolume(u64 pid, u64 delay, float volume) {
+    const struct {
+        float volume;
+        u64 pid;
+        u64 timespan;
+    } in = { volume, pid, 0 };
```

Comment:
0 likely not intended?

Reply:
ah yeah. I just noticed that i didn't have the option to set timespan in any of the functions. Would you like me to add it as one of the args?

btw, i am not 100% sure timespan is correct, i was told by a yuzu dev. It's the delay until it takes effect, although i didn't test values other than 0.

Reply:
oops, i do actually set the delay 😆 i called it delay in the args, but timespan in the struct. ill correct the names